src/models/state/tx_creation_config.rs

```diff
@@ -0,0 +1,145 @@
+use std::fmt::Debug;
+use std::sync::Arc;
+
+use super::tx_proving_capability::TxProvingCapability;
+use super::wallet::address::SpendingKey;
+use super::wallet::utxo_notification::UtxoNotificationMedium;
+use crate::job_queue::triton_vm::TritonVmJobQueue;
+use crate::models::proof_abstractions::tasm::program::TritonVmProofJobOptions;
+
+/// When the selected inputs represent more coins than the outputs (with fee)
+/// where does this change go?
+#[derive(Debug, Clone, Default)]
+pub(crate) enum ChangePolicy {
+    /// If the change is nonzero, crash the transaction creator.
+    #[default]
+    None,
+
+    /// If the change is nonzero, create a new UTXO spendable by this key and
+    /// via this notification medium. Otherwise, do not create a change output.
+    Recover {
+        key: Box<SpendingKey>,
+        medium: UtxoNotificationMedium,
+    },
+
+    /// If the change is nonzero, ignore it.
+    #[cfg(test)]
+    Burn,
+}
+
+/// Options and configuration settings for creating transactions
+#[derive(Debug, Clone, Default)]
+pub(crate) struct TxCreationConfig {
+    change_policy: ChangePolicy,
+    prover_capability: TxProvingCapability,
+    triton_vm_job_queue: Option<Arc<TritonVmJobQueue>>,
+    record_details: bool,
+    proof_job_options: TritonVmProofJobOptions,
+}
+
+impl TxCreationConfig {
+    /// Enable change-recovery and configure which key and notification medium
+    /// to use for that purpose.
+    pub(crate) fn recover_change(
+        mut self,
+        change_key: SpendingKey,
+        notification_medium: UtxoNotificationMedium,
+    ) -> Self {
+        self.change_policy = ChangePolicy::Recover {
+            key: Box::new(change_key),
+            medium: notification_medium,
+        };
+        self
+    }
+
+    /// Configure the proving capacity.
+    pub(crate) fn with_prover_capability(mut self, prover_capability: TxProvingCapability) -> Self {
+        self.prover_capability = prover_capability;
+        self
+    }
+
+    /// Configure which job queue to use.
```

Comment:
"which job queue to use?"

The intent here (multiple queues) appears to be at odds with the job-queue design.

the purpose of the job-queue is/was to have a *single* serialized queue so that proving jobs do not compete with eachother for machine resources (cpu, ram).

Reply:
Then why do we pass around a pointer to the job queue through a bunch of function interfaces? If the purpose is to have a single serialized queue that guarantees that jobs do not compete with each other for resources, then surely a static `OnceLock<TritonVmJobQueue>` is a more fitting solution?

Reply:
It might be more fitting.  The fact remains the raison d'être for the job-queue is and remains serializing jobs running in the VM so as not to exceed available resources.

It's possible we might want to change that at some point, but if so it should be done carefully.